### ISSUE TITLE
Error Data too long when authenticating via Gitea OAuth

### ISSUE THREAD

**Issue by gytisrepecka** [writefreely/writefreely#516]
### Describe the bug
Following error is returned when loggin in via Gitea:

```json
{"error":"Error 1406: Data too long for column 'access_token' at row 1"}
```

### Steps to reproduce
1. Attempt to login to WriteFreely via Gitea OAuth;
2. When redirected from Gitea back to WriteFreely above error is displayed.

### Expected behavior
Should successfully login.

### Application configuration
 - **Single mode or Multi-user mode?** Multi-user
 - **Database?** [mysql/sqlite] MySQL
 - **Open registration?** [yes/no] No
 - **Federation enabled?** [yes/no] Yes

**Version or last commit**: v0.13.1

**Comment by gytisrepecka** [comment]
Problem originates from table `oauth_users` structure:

```sql
CREATE TABLE `oauth_users` (
  `user_id` int(11) NOT NULL,
  `remote_user_id` varchar(128) NOT NULL,
  `provider` varchar(24) NOT NULL DEFAULT '',
  `client_id` varchar(128) NOT NULL DEFAULT '',
  `access_token` varchar(512) NOT NULL DEFAULT '',
  UNIQUE KEY `oauth_users_uk` (`user_id`,`provider`,`client_id`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1
```

Notice `access_token` is defined as `varchar(512)`. While testing OAuth with Gitea server v1.15.3 returned token is longer - 858 positions, therefore need to extend target column.

This query fixes the problem:

```sql
ALTER TABLE `oauth_users`
MODIFY `access_token`
varchar(1024);
```

What would be the best way to include this statement into [migrations](https://github.com/writefreely/writefreely/blob/develop/migrations/migrations.go)?

**Comment by thebaer** [comment]
Thanks for reporting this, @gytisrepecka! To add this migration, you can follow the example of this commit: 5de4d2086bba965333b2a70beeea532e9f2daece

**Comment by gytisrepecka** [comment]
Thanks, I will create a Pull Request.

**Comment by decentral1se** [comment]
Thanks for raising this, ran into it too integrating with Keycloak! Good to see a fix coming down the tubes :heart:

**Comment by p00rt** [comment]
@gytisrepecka if you're creating a PR, could you make the column 2048 chars long? Keycloak gives very long auth tokens (>1300 bytes). Thanks!

**Comment by gytisrepecka** [comment]
> @gytisrepecka if you're creating a PR, could you make the column 2048 chars long? Keycloak gives very long auth tokens (>1300 bytes). Thanks!

Since it's `varchar`, I don't see a problem. Will try it out and create a Pull Request 👍

**Comment by Tealk** [comment]
With authentik there is also the problem, there I had to increase the size to 2048

**Comment by thebaer** [comment]
This should be fixed with #618, which will be in the next release.
